### DISCUSSION TITLE
No highstate has completed since the system was restarted |  Specified SLS 'global' in environment 'base' is not available on the salt master

### DISCUSSION THREAD

**Discussion by Axtrus** [Security-Onion-Solutions/securityonion#12069] [2.4]
### Version

2.4.30

### Installation Method

Security Onion ISO image

### Description

upgrading

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

64

### RAM

252

### Storage for /

2 TB

### Storage for /nsm

2 TB

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

Hi guys,

I was updating my security onion from version 2.4.20 Hotfix [20231012]  to version 2.4.30 Hotfix [20231219] Changes.

And i had internet issues, then i suddenly noticed that all the services were unavailable 
so-status
 **_⌛ System appears to be starting. No highstate has completed since the system was restarted._**

and i get the following message while restarting the salt  master 

 _salt-call state.highstate
[INFO    ] Loading fresh modules for state activity
local:
    Data failed to compile:
----------
    Pillar failed to render with the following messages:
----------
    Specified SLS 'data.*' in environment 'base' is not available on the salt master
----------
    Specified SLS 'zeeklogs' in environment 'base' is not available on the salt master
----------
    Specified SLS 'global' in environment 'base' is not available on the salt master_

Can anyone help me solve this?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment]
Here are some things you can try. 

WARNING! `so-saltstack-update` is an undocumented command and is not used under normal circumstances. It is only being mentioned here since the update failed due to Internet issues. When running `so-saltstack-update`, it is important to specify the correct Github branch for whatever version you're running. For Security Onion 2.4, that is currently `2.4/main`.

```
sudo so-saltstack-update 2.4/main
```

```
sudo so-docker-refresh
```

```
sudo so-checkin
```

If all else fails, you can always re-install.

**Comment by sudesh0sudesh** [comment]
We have also observed same problem for a standalone installtion; If you want we can test it so that we can find the root cause
